autoload/mucomplete.vim

```diff
@@ -302,8 +328,18 @@ endf
 fun! mucomplete#init(dir, tab_completion) " Initialize/reset internal state
   let g:mucomplete_with_key = a:tab_completion
   let s:dir = a:dir
-  let s:compl_methods = get(b:, 'mucomplete_chain',
-        \ get(g:mucomplete#chains, getbufvar("%", "&ft"), g:mucomplete#chains['default']))
+
+  " get the chain of completion methods in this order:
+  " 1. check if a chain is defined for the current syntax group and filetype
+  " 2. try b:mucomplete_chain
+  " 3. finally try g:mucomplete#chains[&ft] or g:mucomplete#chains['default']
+
+  try
+    let s:compl_methods = s:get_chain_for_scope()
+  catch
+    let s:compl_methods = s:get_chain_for_buffer()
+  endtry
```

Comment:
Why is this in a try-catch block?

Reply:
Because at first I thought I could throw an error to go back to the main function. Then I put the normal chains in `get_chain_for_buffer()`, and I returned them from `get_chain_for_scope()`, so it is not necessary.

But it can still protect from dictionary errors and I though it was more readable than
```vim
let s:compl_methods = get_chain_for_scope()
if empty(s:compl_methods)
  let s:compl_methods = s:get_chain_for_buffer()
endif
```

It's mostly cosmetic, though. Should I change it?

Reply:
No need. The code seems to work fine, and I've merged it as-is locally. I will make such changes myself, then push it to master.